halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/services/v1/DeploymentService.java

```diff
@@ -34,20 +38,27 @@
   @Autowired
   ConfigService configService;
 
-  public DeploymentConfiguration getDeploymentConfiguration(String deployment) {
+  public DeploymentConfiguration getDeploymentConfiguration(NodeReference reference) {
+    String deploymentName = reference.getDeployment();
     Halconfig halconfig = configService.getConfig();
     List<DeploymentConfiguration> matching = halconfig.getDeploymentConfigurations()
         .stream()
-        .filter(d -> d.getName().equals(deployment))
+        .filter(d -> d.getName().equals(deploymentName))
         .collect(Collectors.toList());
 
     switch (matching.size()) {
       case 0:
-        throw new IllegalConfigException("The " + NamingService.deployment(deployment) + " not found");
+        throw new IllegalRequestException(new ProblemBuilder(Problem.Severity.FATAL,
+            "No matching deployment could be found")
+            .setReference(reference)
+            .setRemediation("Create a new deployment").build());
       case 1:
         return matching.get(0);
       default:
-        throw new IllegalConfigException("The " + NamingService.deployment(deployment) + " appears more than once");
+        throw new IllegalConfigException(new ProblemBuilder(Problem.Severity.FATAL,
+            "More than one matching deployment found")
```

Comment:
Is this multiple matching deployments of the given name? Could you clarify here where they came from to help track them down?